xcp/pci.py

```diff
@@ -272,6 +273,7 @@ def __init__(self):
             if len(el) == 6:
                 self.devs[el[0]]['subvendor'] = el[4]
                 self.devs[el[0]]['subdevice'] = el[5]
+        cmd.stdout.close()
```

Comment:
Why is this needed? As `cmd` goes out of scope here, the resources should be cleaned up. Are you sure it isn't related to something specific to what the testcase is doing?

In any case, I think using a context manager here would be preferable.

Reply:
@rosslagerwall Yes, sure:

It's a normal warning, everybody who enables PYTHONWARNING (in tests or not) sees:
https://stackoverflow.com/questions/58649679/resourcewarning-unclosed-file-io-bufferedreader-name-4

I think the reason might be that cmd.stdout has an associated open file descriptor and it's cleanup by the subprocess being waited on may not be noticed by the warning.

I named the branch `pci/fix-harmless-unclosed-Popen-warn` so yes, I'm quite certain that it should be harmless, but we can be 100 certain that it doesn't hurt fixing the warning.

In summary, the question:
> Why is this needed?

is best answered by others getting it as well and it doesn't hurt.

https://stackoverflow.com/questions/58649679/resourcewarning-unclosed-file-io-bufferedreader-name-4

And sadly: The `subprocess.Popen.{__enter__,__exit__}()` context manager / `with subprocess.Popen()` is not implemented in Python 2 `subprocess.Popen`: `python2 -c 'import subprocess as s;s.Popen("pwd").__enter__()'`

Also, for the future update to move to context managers for Popen():

- All the `Popen()` test cases using plain `mock` Popen would (if not removed) need to be updated to fake/mock the context manager API and that's not easy to do with mock.

- I've found a library which supports mocking Popen() much better:
    - It correctly mocks the behavior of returning bytes unless text mode is enabled, and also
    - supports using a context manager without having the change all the Popen tests when we can migrate to use context managers for Popen when we break Python2 after we've moved away from it.
    
That means that we can slowly start updating the Popen() testcases, and then later, when we have migrated Dom0 away from Python2, we can complete the migration and drop Python2.

Therefore, as preparation, here is now the start of the way towards this by updating the cmd mock testcase mocking Popen().